lib/datadog/tracing/contrib/rails/framework.rb

```diff
@@ -68,61 +69,49 @@ def self.activate_rack!(datadog_config, rails_config)
             )
           end
 
-          def self.activate_active_support!(datadog_config, rails_config)
+          def self.activate_active_support!(datadog_config)
             return unless defined?(::ActiveSupport)
 
             datadog_config.tracing.instrument(:active_support)
           end
 
-          def self.activate_action_cable!(datadog_config, rails_config)
+          def self.activate_action_cable!(datadog_config)
             return unless defined?(::ActionCable)
 
             datadog_config.tracing.instrument(:action_cable)
           end
 
-          def self.activate_action_mailer!(datadog_config, rails_config)
+          def self.activate_action_mailer!(datadog_config)
             return unless defined?(::ActionMailer)
 
-            datadog_config.tracing.instrument(
-              :action_mailer,
-              service_name: rails_config[:service_name]
-            )
```

Comment:
I suggest we make this a conditional check if `:action_mailer` has already been activated. If so, skip this whole `activate_action_mailer!` method.

This will allow:
```ruby
c.tracing.instrument :rails, service_name: 'my-rails'
c.tracing.instrument :action_mailer, service_name: 'my-mailer'
```
to set the `action_mailer` service name to the desired value, **while keeping backwards compatibility**.